src/utils.coffee

```diff
@@ -42,12 +44,26 @@ parse = (rawLink) ->
 # Read file to string
 readFile = (filename) ->
   try
-    content = (fs.readFileSync filename).toString()
-    return content
+    validUrlRegex = /(http|https):\/\/(\w+:{0,1}\w*)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%!\-\/]))?/
+    if validUrlRegex.test(filename)
+      done = false
+      content = null
+      request(filename, (error, response, body) ->
+        if !error && response.statusCode == 200
+          content = body
+          done = true
+        else
+          done = true
+      )
+      # makes it non-blocking and synchronous.
+      require('deasync').loopWhile( -> !done)
```

Comment:
At the moment the library interface for hercule is async as is `inflate` (in preparation for HTTP support) so the function to retrieve the HTTP file contents can be async.